### ISSUE TITLE
Hitting tab to indent replaces content with a completion

### ISSUE THREAD

**Issue by ssigwart** [microsoft/vscode#135823]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.61.2 (I've seen this before this version)
- OS Version: MacOS Big Sur

Steps to Reproduce:

1.  Create a typescript file like this:
  ```typescript
if (true)
{
let c = 1;
let d = 1;
}
  ```
2.  Select from just after c to just after d.
3. Hit tab to try to indent

The result is this:
```typescript
if (true)
{
let c = 1;
let do {
	
} while (condition); = 1;
}
```

This seems to trigger completions.  Other example are:

### Select from after c to after e in `let d`
```typescript
if (true)
{
let c = 1;
lengtht d = 1;
}
```

### Select from after c to end of next line
```typescript
if (true)
{
let cc
}
```

Note that this seems to depend on what your recent completions are.  For example, that last one will become `let cconsole` if the last auto-complete you accepted was `console`.

**Comment by ssigwart** [comment]
@rebornix, I have a little more info.  The build from source had `editor.tabCompletion` off.  When I enabled that, I get the same issue.  Another example is with PHP and hitting tab with this selection.
![image](https://user-images.githubusercontent.com/1426848/140611892-3bff2355-ae69-49e3-bfa6-6a5f1496e51d.png)
### Result
![image](https://user-images.githubusercontent.com/1426848/140611909-775f878e-912d-4b49-a364-79e16d867fb1.png)

### Code
```php
<?php
$e = new Exception();
echo $e->getFile();
```

**Comment by ssigwart** [comment]
@rebornix, does this still need the "needs more info" label?  I think the Nov 6 comment should help to duplicate it.  I have a fix in #136572.

**Comment by ssigwart** [comment]
Yes, this is still an issue.  @rebornix, PR #136572 has a merge conflict now.  Please let me know if you want me to rebase it before you review it.

**Comment by aiday-mar** [comment]
Hi I am currently not able to reproduce this. Do you still see this issue?

**Comment by ssigwart** [comment]
@aiday-mar, yes, I do. I just tried on the insider version (1.96), but don't see it there. I just tried started the stable version with extensions disabled and I still see it happening.

I did so testing and found out that I can duplicate in Insider too if I set `editor.tabCompletion` to on.

**Comment by ssigwart** [comment]
@aiday-mar, I also wanted to note that I have PR #136572 that has a fix for this.

**Comment by aiday-mar** [comment]
cc @hediet concerns inline completions
